### ISSUE TITLE
[BUG] Data race in consensus tests

### ISSUE THREAD

**Issue by gokutheengineer** [pokt-network/pocket#630]
## Objective

There is a data race bug in consensus tests.

## Origin Document

Running race tests produces the data race findings, example tests:

- `go test -count=1 -timeout 30s -v -race -run ^TestHotstuff github.com/pokt-network/pocket/consensus/e2e_tests`
- `go test -count=1 -timeout 30s -v -race -run ^TestPacemaker github.com/pokt-network/pocket/consensus/e2e_tests`

## Goals

- Solve data race happenining in consensus tests.

## Deliverable

- [ ] Consensus data race tets doesn't result with any data race.

## Non-goals / Non-deliverables

- Change in business logic.

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @gokutheengineer 
**Co-Owners**: @Olshansk @deblasis

**Comment by Olshansk** [comment]
@gokutheengineer I think adding a screenshot of the error would really help

**Comment by Olshansk** [comment]
This was done as one of the commits pushed into a different PR to unblock work: https://github.com/pokt-network/pocket/pull/528/commits
